### PR TITLE
[IMPROVEMENT] Remove own typing indicator

### LINKED ISSUE #1242: [IMPROVEMENT] Remove own typing indicator
When you type using e.g. web client and have the same channel open on mobile, the typing indicator shows your own username. We may hide this.

### PR DESCRIPTION
**Pull request by soundstorm**
Closes #1242

@RocketChat/ios

We still need tests for most of the ViewControllers :/

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -512,10 +512,13 @@ final class ChatViewController: SLKTextViewController {
 
     func registerTypingEvent(_ subscription: Subscription) {
         typingIndicatorView?.interval = 0
+        let user = AuthManager.currentUser()
 
         SubscriptionManager.subscribeTypingEvent(subscription) { [weak self] username, flag in
             guard let username = username else { return }
 
+            if username == user?.username { return }
```
Comment (artrmz) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Could be added to the guard statement as well I guessl, would be a little bit cleaner ;)

Comment (soundstorm) [reply Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Yep :)

File: Rocket.Chat/Managers/Model/AuthManager.swift [hunk 2]
```diff
@@ -326,7 +326,7 @@ extension AuthManager {
     static func auth(credentials: OAuthCredentials, completion: @escaping MessageCompletion) {
         let params = [
             "oauth": [
-                "credentialToken": credentials.token ?? "",
+                "credentialToken": credentials.token,
```
Comment (rafaelks) [inline Rocket.Chat/Managers/Model/AuthManager.swift]: Why this change?

Comment (soundstorm) [reply Rocket.Chat/Managers/Model/AuthManager.swift]: https://github.com/RocketChat/Rocket.Chat.iOS/blob/bcf7855ace5a20249dfa51f4431f51b5dd680dc4/Rocket.Chat/Managers/OAuthManager.swift#L15
@rafaelks token is non optional, so this just gives a warning in Xcode, that `?? ""` is not needed.

Comment (rafaelks) [reply Rocket.Chat/Managers/Model/AuthManager.swift]: Sounds good! :-)

**Comment by rafaelks** [comment]
Thank you @soundstorm!

**Comment by rafaelks** [comment]
@soundstorm I'm still experiencing this issue here when I type in other device (web or iPad) and then I see on my iPhon that I'm typing.

**Comment by soundstorm** [comment]
@rafaelks uhm, okay, I'll check.

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift [hunk 3]
```diff
@@ -512,9 +512,10 @@ final class ChatViewController: SLKTextViewController {
 
     func registerTypingEvent(_ subscription: Subscription) {
         typingIndicatorView?.interval = 0
+        guard let user = AuthManager.currentUser() else { return Log.debug("Could not register TypingEvent") }
 
         SubscriptionManager.subscribeTypingEvent(subscription) { [weak self] username, flag in
-            guard let username = username else { return }
+            guard let username = username, username != user.username else { return }
```
Comment (soundstorm) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: @rafaelks these changes got reverted by #1048 so not my fault :(

Comment (rafaelks) [reply Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Ouch! Sorry about that! Let's include it again. 👍
